### ISSUE TITLE
confidentialInstanceConfig fields required/optional swap in next major version update

### ISSUE THREAD

**Issue by deeglaze** [hashicorp/terraform-provider-google#17090]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment. If the issue is assigned to the "modular-magician" user, it is either in the process of being autogenerated, or is planned to be autogenerated soon. If the issue is assigned to a user, that user is claiming responsibility for the issue. If the issue is assigned to "hashibot", a community member has claimed the issue already.

### Description

The confidential_instance_config field for compute has an attribute, "enable_confidential_compute" we wish to deprecate in favor of a new more explicit enum attribute, "confidential_instance_type" that selects the confidential computing technology for the instance.

Right now, enable_confidential_compute is required, but we want to make it optional.
With https://github.com/GoogleCloudPlatform/magic-modules/pull/9847, confidential_instance_type is optional, but we want to make it required.

This is a breaking change, so it would require a major version update.

### New or Affected Resource(s)

This effects the compute product's instance creation logic in a single instance, instance templates, and regional instance templates.

* confidential_instance_config
   - enable_confidential_compute
   - confidential_instance_type

* google_compute_instance_template
* google_compute_instance

### Potential Terraform Configuration

```tf
resource "google_compute_instance_template" "foobar" {
  confidential_instance_type = "SEV"
}
```
instead of either
```tf
resource "google_compute_instance_template" "foobar" {
  enable_confidential_compute = true
  confidential_instance_type = "SEV"
}
```
or
```tf
resource "google_compute_instance_template" "foobar" {
  enable_confidential_compute = true
}
```

and furthermore
```tf
resource "google_compute_instance_template" "foobar" {
  confidential_instance_type = "SEV_SNP"
}
```
instead of
```tf
resource "google_compute_instance_template" "foobar" {
  enable_confidential_compute = true
  confidential_instance_type = "SEV_SNP"
}
```
### References

https://github.com/hashicorp/terraform-provider-google/issues/14768
https://github.com/GoogleCloudPlatform/magic-modules/pull/9847

b/340524527

**Comment by c2thorn** [comment]
@roaks3 this didn't seem to get forwarded, do we need to do anything extra here?

**Comment by roaks3** [comment]
Ah yea, it looks like a service label didn't get added. Our GitHub Action looks in the "New Resources" section for resources that match [this file](https://github.com/GoogleCloudPlatform/magic-modules/blob/main/tools/issue-labeler/labeler/enrolled_teams.yml), and didn't find any here.

I updated the description to include the resources I saw described, and manually added the service label, but please update if I've done something wrong.

**Comment by karolgorc** [comment]
I don't think this is a good change. Currently these fields are 1:1 representation of the API. This type of configuration doesn't throw an error.
![image](https://github.com/user-attachments/assets/2cc48277-7f76-4d96-a095-af68f9de61fe)

If it did then we could go for changing it without user interaction but currently this would restrict some of API functionality and could cause problems with future changes

**Comment by c2thorn** [comment]
on the
